src/components/Image/index.js

```diff
@@ -10,20 +11,24 @@ class Image extends React.Component {
     constructor(props) {
         super(props);
 
+        this.debouncedConfigureImageSource = _.debounce(this.configureImageSource, 220);
+
```

Comment:
This is required only for attachments flow, should this debounce be set here? 

As it add debounce all other views which uses this component

Reply:
It seems fine to debounce these in the image component since the logic to configure the image source is in it already.

Reply:
Hmm i don't mind the debounce but can someone remind me why we need the debounce here? Is it because if the user clicks next rapidly we want to load the image properly or is there some other reason?

I'm mostly wondering if we can trigger the debounce immediately via,
```
this.debouncedConfigureImageSource = _.debounce(this.configureImageSource, 220, true);
```
So that non-attachments are not impacted and attachments should load faster without the delay. Wouldn't that work better?

Reply:
That could work if there’s a Boolean that would make that false for the attachment carousel

Reply:
I tested this a bit more wondering if it's related to clicking next button rapidly or resizing and i could not reproduce either when I remove the debounce (tested on web and ios). Can we recheck if this debounce is necessary and if so maybe add a comment explaining why?

Reply:
I believe this was purely an optimization for clicking too fast

Reply:
Can you retest without debounce and check if clicking too fast creates any issues? If not then I think let's not have debounce unless it's really necessary.

Reply:
@chiragsalian I think it does cause an issue when I remove it. You'll see that when I click fast the previous images flash like it's catching up. Currently it's just the loading spinner reseting a few times.

https://user-images.githubusercontent.com/6119181/223578335-5143db8d-74a6-4ca9-b694-6e943bccecf9.mov

Reply:
Cool, and what about running debounce immediately, does it work fine or does it have the same issue?
```
this.debouncedConfigureImageSource = _.debounce(this.configureImageSource, 220, true);
```

Reply:
I don't see anything wrong with that. I guess what happens is the first call goes through then subsequent calls are debounced preventing the catch up issue, while allowing images that don't require this to load in time. I'm going to push the changes.

Reply:
I'm still seeing it when I do that.

https://user-images.githubusercontent.com/6119181/223586194-10ec8c31-0a34-46a5-9a1d-d246eb46efd4.mov

Reply:
cool, well it's good enough for me. I'm fine with the debounce. Can you investigate doing,
> That could work if there’s a Boolean prop that would make that false for the attachment carousel

such that,
```
_.debounce(this.configureImageSource, 220, shouldDebounceImmediate);
```
shouldDebounceImmediate is true by default but just for attachment carousel its false. I think it would be a nice addition. Thoughts?

Reply:
Hey @chiragsalia, @JediWattson, @luacmartins, it seems this debounce is an internal detail only related to the carousel functionality - it shouldn't have been put here

And it also seems to be a workaround - what's causing the images to flicker between the old and then new a few times. I believe we can investigate and solve the root problem 
- if the flicker happens on all platforms - it's probably an App issue
- if it's only on web - then it's probably a problem in `react-native-web` Image component

Also because of the debounce we would never be able to instantly load images from cache, loading an image would never take less than 220ms 
- https://github.com/Expensify/App/issues/15922

---

BTW another PR is getting entirely getting rid of `state` (adding `state` here was kind-a not needed)
- https://github.com/Expensify/App/pull/15663

Reply:
Makes sense, looking back again it was added to address a crash issue mentioned [here](https://github.com/Expensify/App/pull/9279#issuecomment-1396319588).

Reply:
We just need to reinvestigate this again and decide on this @JediWattson.
cc: @chiragsalian @luacmartins

Reply:
We don't have enough (History is messed up) information on when or why these changes are done. So it's better to revert this change alone. As I remember it was for flickering for images, but If you know why please let us know @JediWattson thanks!

Reply:
Removed the debounce and couldn't recreate the clicking fast problem (on main)

Reply:
@kidroca I had found [this occurred](https://github.com/Expensify/App/pull/9279#discussion_r1128734946) when you completely remove the debounce. I'm still seeing it, but it's subtle. My thinking was to drill down a prop that identifies that the image component is in the attachment carousel and use that as a boolean to set the immediate flag arg in the debounce. It'll be inverse logic too such as: 

```
this.debouncedConfigureImageSource = _.debounce(this.configureImageSource, 220, !props.isAttachmentImage);
```

if `props.isAttachmentImage` is true it'll turn false, but for the rest of the images it'll be true from being undefined

Reply:
There should be no debounce 
1. It's a workaround 
2. From your explanation the change is only concerning the carousel - everything else would not use debounce (the same way the carousel will)

I've submitted a PR for RN web to our fork. If fixes some inconsistencies with the Image component. I think it would address the need for debounce as well

Reply:
That definitely tracks. I don't know if the way I'm going about it is the same as what you mentioned, but I was digging into this more and the crux of the problem is that `getSize` really shouldn't be used at all. In fact I managed to refine the Image component into a few lines of code.

```
function Image({ source, isAuthTokenRequired, ...props }) {   
    if (isAuthTokenRequired) {
        source.uri = addEncryptedAuthTokenToURL(source.uri);
    }
    
    // eslint-disable-next-line
    return <RNImage {...props} source={source} />;
}
```

There's some changes I had to make in a couple other places, but not too much is really changed. I'm now using `onLoad` to send the height and width which changes the arguments to the parent functions. Also I needed a check in the ImageView to make sure the url was correct before setting the scale. 

I was wondering if all this could be a proposal for [this issue](https://github.com/Expensify/App/issues/15922) because it does reduce the loading? I just need to clean up the code a bit to make a PR for it too.

Reply:
The custom `onLoad` with `getSize` was also a workaround, because at the time `react-native-web` did not return image dimensions with the `onLoad` callback prop

Our fork of RN web has a fix for this and I'm applying it to the Image component, but the PR is blocked ATM
- https://github.com/Expensify/App/pull/13036

Reply:
@kidroca  it looks like a change for removing the debounce was merged which should resolve this thread. Was there anything else to consider or is this all set??